includes/abstracts/abstract.llms.post.model.php

```diff
@@ -910,70 +911,130 @@ protected function scrub_field( $val, $type ) {
 	 *
 	 * @since 3.0.0
 	 * @since 3.30.3 Use `wp_slash()` when setting properties.
+	 * @since [version] Turned to be only a wrapper for the update_model() method.
 	 *
-	 * @param string $key Key of the property.
-	 * @param mixed  $val Value to set the property with.
+	 * @param string|array $key_or_array Key of the property or a an associative array of key/val pairs.
+	 * @param mixed        $val          Optional. Value to set the property with. Default empty string.
+	 *                                   This parameter will be ignored when the first parameter is an associative array of key/val pairs.
 	 * @return boolean true on success, false on error or if the submitted value is the same as what's in the database
 	 */
-	public function set( $key, $val ) {
+	public function set( $key_or_array, $val = '' ) {
 
-		$val = $this->scrub( $key, $val );
+		$model_array = array();
+		if ( ! is_array( $key_or_array ) ) {
+			$model_array = array(
+				$key_or_array => $val,
+			);
+		} else {
+			$model_array = $key_or_array;
+		}
+		return $this->update_model( $model_array );
 
-		// update WordPress Post Properties using the wp_insert_post() function
-		if ( in_array( $key, array_keys( $this->get_post_properties() ) ) ) {
+	}
 
-			$post_key = 'post_' . $key;
 
-			switch ( $key ) {
+	/**
+	 * Updater
+	 *
+	 * @since [version]
+	 *
+	 * @param array $model_array Associative array of key/val pairs.
+	 * @return boolean true on success, false on error or if there was nothing to update.
+	 */
+	public function update_model( $model_array ) {
 
-				case 'content':
-					$val = apply_filters( 'content_save_pre', $val );
-				break;
+		if ( empty( $model_array ) ) {
+			return false;
+		}
 
-				case 'excerpt':
-					$val = apply_filters( 'excerpt_save_pre', $val );
-				break;
+		$llms_post = array(
+			'post' => array(),
+			'meta' => array(),
+		);
 
-				case 'menu_order':
-					$post_key = 'menu_order';
-				break;
+		$post_properties = array_keys( $this->get_post_properties() );
+		$unsettable_properties = $this->get_unsettable_properties();
 
-				case 'title':
-					$val = apply_filters( 'title_save_pre', $val );
-				break;
+		foreach ( $model_array as $key => $val ) {
 
-			}
+			$val = $this->scrub( $key, $val );
```

Comment:
I'd like to see the content of this loop be broken out into a protected method solely responsible for filtering a single key when it's being updated.

That way the content of the loop can be reduced to something like:

```php
$val = $this->apply_update_filters( $this->scrub( $key, $val ), $key );
```

I think this should reduce the overall complexity and logic of this method a bit

Maybe I'm overcomplicating it though? Thoughts?

Reply:
Mmm sorry I don't get this :) 
...Or maybe yes because I though about that too, but basically I made a method that wrapped this:
https://github.com/gocodebox/lifterlms/blob/565950b04461d9aaca4f66852ea03de2ad8c2c10/includes/abstracts/abstract.llms.post.model.php#L960-L993
without the `continue` and ending with:
`return apply_filters( 'llms_set_' . $this->model_post_type . '_' . $key, $val, $this );`
in place of
`$llms_post[ $type ][ $llms_post_key ] = apply_filters( 'llms_set_' . $this->model_post_type . '_' . $key, $val, $this );`

But then I realized that in the `foreach` I still needed to perform the same checks:
```php
if ( in_array( $key, $post_properties ) ) {
...
} elseif ( ! in_array( $key, $unsettable_properties ) ) {
...
} else {
	continue;
}
```
because I needed to separate the `post_properties` from the `meta_properties`.
So I basically gave up because of that repetition.
But I can see how creating that method would lighten the reading/understanding of the code.
So if I got what you mean and you're fine with the "checks repetition" (or if you have a good way to avoid it that I'm totally missing) let me know and I'll do it!

Reply:
`apply_update_filters` implemented in ea1221f3fa2ea8fc1f6ae3027772283cb73fb89a

What do you think?! Still a little bit too cryptic don't you think?

Reply:
@eri-trabiccolo man I'm sorry... Now that i'm seeing it this way you're right it was better the other way as there's too much redundancy now. Can you rollback to your previous version fo this function (without breaking out the `apply_update_filter()` method) and I'll have another look and merge it.

Sorry!

Reply:
:D No problem at all. 
Latest commit reverted and force pushed.